mne/preprocessing/ica.py

```diff
@@ -842,9 +828,10 @@ def _transform_raw(self, raw, start, stop, reject_by_annotation=False):
             raise RuntimeError('No fit available. Please fit ICA.')
         start, stop = _check_start_stop(raw, start, stop)
 
-        picks = pick_types(raw.info, include=self.ch_names, exclude='bads',
-                           meg=False, ref_meg=False)
-        if len(picks) != len(self.ch_names):
+        picks = _picks_to_idx(
+            raw.info, self.ch_names, exclude=[], allow_empty=True)
+        raise_ = False if len(picks) == len(self.ch_names) else True
+        if raise_:
```

Comment:
I think it was simpler before... I'm assuming you have this `raise_` just for debugging, or it's cruft from when this was a more complicated conditional or something?
```suggestion
        if len(picks) != len(self.ch_names):
```